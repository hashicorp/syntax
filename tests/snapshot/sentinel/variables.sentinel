# https://developer.hashicorp.com/sentinel/docs/language/variables

a = 1       // a = 1
b = a       // b = 1
a = "value" // a = "value", b = 1
c = a       // c = "value", b = 1

a = c // Error!
c = 1

s = "1.1"
a = int(s)   // a = 1
a = float(s) // a = 1.1

a = 1
s = string(a) // s = "1"
