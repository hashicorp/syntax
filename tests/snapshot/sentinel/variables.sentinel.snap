># https://developer.hashicorp.com/sentinel/docs/language/variables
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.sentinel comment.line.number-sign.sentinel
>
>a = 1       // a = 1
#^^ source.sentinel
#  ^ source.sentinel keyword.operator.symbol.sentinel
#   ^ source.sentinel
#    ^ source.sentinel constant.numeric.number.sentinel
#     ^^^^^^^ source.sentinel
#            ^^^^^^^^ source.sentinel comment.line.double-slash.sentinel
>b = a       // b = 1
#^^ source.sentinel
#  ^ source.sentinel keyword.operator.symbol.sentinel
#   ^^^^^^^^^ source.sentinel
#            ^^^^^^^^ source.sentinel comment.line.double-slash.sentinel
>a = "value" // a = "value", b = 1
#^^ source.sentinel
#  ^ source.sentinel keyword.operator.symbol.sentinel
#   ^ source.sentinel
#    ^ source.sentinel string.quoted.double.untitled
#     ^^^^^ source.sentinel string.quoted.double.untitled
#          ^ source.sentinel string.quoted.double.untitled
#           ^ source.sentinel
#            ^^^^^^^^^^^^^^^^^^^^^ source.sentinel comment.line.double-slash.sentinel
>c = a       // c = "value", b = 1
#^^ source.sentinel
#  ^ source.sentinel keyword.operator.symbol.sentinel
#   ^^^^^^^^^ source.sentinel
#            ^^^^^^^^^^^^^^^^^^^^^ source.sentinel comment.line.double-slash.sentinel
>
>a = c // Error!
#^^ source.sentinel
#  ^ source.sentinel keyword.operator.symbol.sentinel
#   ^^^ source.sentinel
#      ^^^^^^^^^ source.sentinel comment.line.double-slash.sentinel
>c = 1
#^^ source.sentinel
#  ^ source.sentinel keyword.operator.symbol.sentinel
#   ^ source.sentinel
#    ^ source.sentinel constant.numeric.number.sentinel
>
>s = "1.1"
#^^ source.sentinel
#  ^ source.sentinel keyword.operator.symbol.sentinel
#   ^ source.sentinel
#    ^ source.sentinel string.quoted.double.untitled
#     ^^^ source.sentinel string.quoted.double.untitled
#        ^ source.sentinel string.quoted.double.untitled
>a = int(s)   // a = 1
#^^ source.sentinel
#  ^ source.sentinel keyword.operator.symbol.sentinel
#   ^ source.sentinel
#    ^^^ source.sentinel support.function.builtin.sentinel
#       ^^^^^^ source.sentinel
#             ^^^^^^^^ source.sentinel comment.line.double-slash.sentinel
>a = float(s) // a = 1.1
#^^ source.sentinel
#  ^ source.sentinel keyword.operator.symbol.sentinel
#   ^ source.sentinel
#    ^^^^^ source.sentinel support.function.builtin.sentinel
#         ^^^^ source.sentinel
#             ^^^^^^^^^^ source.sentinel comment.line.double-slash.sentinel
>
>a = 1
#^^ source.sentinel
#  ^ source.sentinel keyword.operator.symbol.sentinel
#   ^ source.sentinel
#    ^ source.sentinel constant.numeric.number.sentinel
>s = string(a) // s = "1"
#^^ source.sentinel
#  ^ source.sentinel keyword.operator.symbol.sentinel
#   ^ source.sentinel
#    ^^^^^^ source.sentinel support.function.builtin.sentinel
#          ^^^^ source.sentinel
#              ^^^^^^^^^^ source.sentinel comment.line.double-slash.sentinel
>